src/shared/utils/duration.ts

```diff
@@ -166,8 +173,9 @@ export const getTimeRangeLabel = (
     return timeRange.lower
   }
   if (timeRange.type === 'custom') {
-    const lower = moment(timeRange.lower).format(TIME_RANGE_FORMAT)
-    const upper = moment(timeRange.upper).format(TIME_RANGE_FORMAT)
+    const formatter = createDateTimeFormatter(TIME_RANGE_FORMAT, timeZone)
+    const lower = formatter.format(new Date(timeRange.lower))
+    const upper = formatter.format(new Date(timeRange.upper))
     if (singleDirection && singleDirection === TimeRangeDirection.Upper) {
       return upper
     } else if (
```

Comment:
i know you didn't write this, but while we are making changes to this file... can we also add one more change?

`singleDirection && singleDirection === TimeRangeDirection.Lower` is too wordy. The first part is not necessary. If `singleDirection` is `undefined` then it won't equal to our comparator.

Reply:
Yeah absolutely! I can make that change 👍

Reply:
There's one more below it that i couldn't quote.

Reply:
I missed it the first time too! changed ✔️